### DISCUSSION TITLE
Skip draw event for objects that are not on the screen

### DISCUSSION THREAD

**Discussion by lunat1k-12** [replit/kaboom#473] [Q&A] [answered]
My game became a bit laggy because of the level size. Kaboom performs a draw event for each loaded object. How can I skip this action for those objects which are not on the screen but not destroy them?

**Comment by lajbel** [comment, marked as answer]
you could not draw them at first, and then use the function obj.use(sprite()) when the tiles should be drawn, and then, use obj.hidden = true to hide them. For this, you can assign a Tag and use every() in a onUpdate()

**Reply by lunat1k-12** [reply]
It works, thanks!

**Comment by slmjkdbtl** [comment]
Now it's best to use the `offscreen({ hide: true })` component to not draw objects that's not on screen
